include/modem/location.h

```diff
@@ -267,6 +267,16 @@ struct location_gnss_config {
 	 * @note Only supported with modem firmware v1.3.2 or later.
 	 */
 	bool visibility_detection;
+
+	/**
+	 * @brief Enable GNSS priority mode if GNSS does not get enough runtime.
```

Comment:
There are two different situations which should be clearly separated:

- GNSS is totally blocked and does not get runtime at all (RRC connected mode).
- GNSS may be getting almost all runtime, but it's getting constantly interrupted by LTE (RRC idle mode).

It should be clear that this feature is for the latter case, thus I would rephrase the "does not get enough runtime" part.

Reply:
I was wondering what are the risks when enabling this? Is there a risk of missing important LTE data? i.e. mqtt cloud connection events or even the a-gps data? If yes, should this feature be used only when not (yet) connected (not registered and no PDN active) to LTE? 
Or, enable it only when no a-gps/p-gps phase currently on going?

Reply:
It seems that the priority time window is short enough (40 seconds) that no data is lost. I tested this by triggering MO data (ping) from from Mosh right after requesting the priority window. The ping got sent successfully upon termination of the priority window. I tested MT data similarly with Contabo by scheduling a downlink data transmission to take place right after starting the priority mode. The data was received successfully when priority mode ended.

Reply:
OK, good that ping is getting through and back, but that is only 28 bytes ICMP packet. I was concerned of actual longer cloud events and actual a-gps data, received MT data. I think that it is not guaranteed that the network is buffering that during window of priority mode and it might get lost, thus was thinking if usage of priority window could be limited to non-LTE phase.

Reply:
That is a good point. It did not occur to me that the network might not be willing to buffer larger amounts of data. However, the timing is such in location lib that A-GPS and/or P-GPS data are usually downloaded first even before starting GNSS and therefore the priority mode should not interfere. @tokangas can confirm if this is still true as he was fiddling with the timing yesterday. 

That being said, I don't know if we can guarantee 100% that no data will be lost. On the other hand, the drawback of starting priority mode only when e.g. not registered to network would limit the usability of the feature.